src/simple_ai/models.py

```
@@ -166,24 +166,28 @@ def stream(
         )
 
 
-def select_model_type(model_interface: str = "gRPC", task: str = "complete"):
+def select_model_type(model_interface: str, task: str):
     if model_interface == "gRPC":
         if task == "embed":
             return RpcEmbeddingLanguageModel
         if task == "chat":
             return RpcChatLanguageModel
-        return RpcCompletionLanguageModel
-    return RpcCompletionLanguageModel
+        if task == "complete":
+            return RpcCompletionLanguageModel
+        raise ValueError(
+            f"`task` value must be either `chat`, `embed`, or `complete`, got `{task}`."
+        )
+    return ValueError(f"`model_interface` value must be `gRPC`, got `{task}` instead.")
 
 
 def get_model(model_id: str, metadata: dict = MODELS_ZOO, task: str = "complete"):
     if model_id in metadata.keys():
-        model_interface = metadata.get(model_id).get("network", dict())
-        model_url = model_interface.get("url", None)
-        model_interface = model_interface.get("type", None)
+        network_parameters = metadata.get(model_id).get("network", dict())
+        model_url = network_parameters.get("url", None)
+        model_interface = network_parameters.get("type", None)
```

Think this may result in some difficult to debug situations. eg trying to figure out why the exception on line 180 is saying `model_interface value must be gRPC, got None instead."`

Could either implement schema validation for the config or raise errors here rather than silently using `None` if schema is not met. [Pydantic](https://docs.pydantic.dev/usage/dataclasses/) is good for this

---

Yeah I'm [already using it here](https://github.com/lhenault/simpleAI/blob/main/src/simple_ai/api_models.py). Would be a good idea and cleaner to also use it for the config, didn't think about it tbh but that makes sense.

---

Done something using your suggestion to leverage Pydantic. It does look better and easier to maintain / extend. Let me know your thoughts. :)

---

Nice! looks good :)